compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -1079,8 +1079,8 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             })
             .map_err(|terr| {
                 SignatureMismatch(Box::new(SignatureMismatchData {
```

Comment:
is that sometimes used with an uninstantiated binder?

Reply:
nope, it's always instantiated

Reply:
then we can remove the `Binder` from `SignatureMismatchData` :thinking:

Reply:
isn't that what this PR is doing? 🤔

Reply:
?????????????????????????????????????????????? I might have been utterly confused

god knows what i was thinking. sorry